static/js/reducers/dashboard.js

```diff
@@ -34,7 +35,13 @@ export const dashboard = (state: DashboardsState = INITIAL_DASHBOARDS_STATE, act
   case REQUEST_DASHBOARD: // eslint-disable-line no-case-declarations
     let newBaseState = R.dissoc(username, state);
     if (action.payload === true) {
-      return updateStateByUsername(newBaseState, username, INITIAL_DASHBOARD_STATE);
+      return updateStateByUsername(
+        newBaseState,
+        username,
+        _.merge({}, INITIAL_DASHBOARD_STATE, {
+          noSpinner: true
```

Comment:
We should have `fetchStatus: FETCH_PROCESSING` here too, and because the two branches of this if statement are similar, you can just set `noSpinner: action.payload`.

Regarding your other comment on the dashboard spinner still spinning, we should change that code to also check `noSpinner`